FirebaseInAppMessaging/Sources/Flows/FIRIAMDisplayExecutor.m

```diff
@@ -232,13 +236,25 @@ - (void)impressionDetectedForMessage:(FIRInAppMessagingDisplayMessage *)inAppMes
     // Displayed long enough to be a valid impression.
     [self recordValidImpression:_currentMsgBeingDisplayed.renderData.messageID
                 withMessageName:_currentMsgBeingDisplayed.renderData.name];
+
+    if ([self shouldTrackConversionsOnImpressionForCurrentInAppMessage:_currentMsgBeingDisplayed]) {
+      [self.analyticsEventLogger
+          logConversionTrackingEventForCampaignID:_currentMsgBeingDisplayed.renderData.messageID];
+    }
   } else {
     FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM400011",
                 @"A test message. Record the test message impression event.");
     return;
   }
 }
 
+- (BOOL)shouldTrackConversionsOnImpressionForCurrentInAppMessage:
```

Comment:
Shouldn't this also include that the message should have conversion event defined?

If it doesn't we don't need to track it.

Reply:
We weren't checking on a conversion event previously, from [comments](https://osscs.corp.google.com/firebase-sdk/firebase-ios-sdk/+/master:FirebaseInAppMessaging/Sources/Analytics/FIRIAMAnalyticsEventLoggerImpl.m;drc=b41a1273368e8af2af26d1dfcb135fc2307b94b3;l=126) (and piecing together the code) it looks like we currently always track the conversion and then presumably handle reporting on the backend. Do you think we need to check for this now?

Reply:
We're waiting on Analytics to confirm how to proceed.